azerothcore/azerothcore-wotlk
# feat(Core): CanAccountCreateCharacter hook

## Changes Proposed:
-  Adds a hook to allow modules to programmatically allow or restrict creation of characters of given race, class, etc. per account based on the account's progress.

## Issues Addressed:


## SOURCE:


## Tests Performed:

- Builds, works with test script


## How to Test the Changes:


1. Ensure character creation still works with hook added
2. Ensure hook can prevent or allow character creation based on hooked script

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
